### ISSUE TITLE
hardhat-etherscan breaks on hardhat/console.sol

### ISSUE THREAD

**Issue by ldub** [NomicFoundation/hardhat#1117]
when running 

```
npx hardhat verify --network mainnet
```

i was getting the error

```
[Error: ENOENT: no such file or directory, open '/Users/lev/dev/ldub/token-2/artifacts/build-info/4e7389b25e253b27f945b011dc8f896b.json'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/Users/lev/dev/ldub/token-2/artifacts/build-info/4e7389b25e253b27f945b011dc8f896b.json'
}
```

this was being thrown by `const buildInfo = await artifacts.getBuildInfo(fqName);` on line https://github.com/nomiclabs/hardhat/blob/master/packages/hardhat-etherscan/src/solc/bytecode.ts#L97 when `fqName` was `hardhat/console.sol:console`. 

i added

```
if (fqName === "hardhat/console.sol:console") continue
```

in `token-2/node_modules/@nomiclabs/hardhat-etherscan/dist/src/solc/bytecode.js` and then the script verified just fine 

I am new to hardhat and dont know the root cause of why its even looking at the console but i hope this is sufficient information for @ItsNickBarry or someone else to fix :)

Thanks for the awesome tools all!

**Comment by ldub** [comment]
i should note that i used the `hardhat-dependency-compiler`  plugin and it could be that plugin's bug

**Comment by alcuadrado** [comment]
Hey @ldub, can you create a repo to reproduce this? Thanks!

**Comment by SirLancelot-OG** [comment]
For anyone running into the same issue, `npx hardhat clean` resolve the problem for me.

**Comment by fvictorio** [comment]
Closing this for lack of more info. Please re-open if you can give us instructions on how to reproduce it.

**Comment by mindqaugas** [comment]
I had the smae issue "npx hardhat clean" resolved my problem

**Comment by ru19wu06** [comment]
@SirLancelot-OG  help me a lot!!

**Comment by MZubairLohar** [comment]
> I had the smae issue "npx hardhat clean" resolved my problem

it worked for me as well ! thanks dude

**Comment by alcuadrado** [comment]
I'm reopening this issue and it seems to be pretty common.

**Comment by Decentra-Ecosystem** [comment]
> For anyone running into the same issue, `npx hardhat clean` resolve the problem for me.

Thank you thank you thank you!!!!

**Comment by chefpickle** [comment]
`npx hardhat clean` works for me . thank you @SirLancelot-OG

**Comment by fvictorio** [comment]
I think I know how to reproduce this:

1) Create the basic sample project
2) Install `@nomiclabs/hardhat-etherscan` and add it to the hardhat config. 
3) Add a valid `rinkeby` network and a valid etherscan API key to the config.
4) Deploy the contract to Rinkeby `hh run --network rinkeby scripts/sample-script.js`
5) Go to `Greeter.sol` and rename the contract `Greeter` to `Foo`. **Do not compile the project**.
6) Try to verify the file with `hh verify --network rinkeby <address>`

**Comment by magic990619** [comment]
`npx hardhat clean` works for me. thanks a lot.

**Comment by Huge** [comment]
Hello, new year here, so glad I've found the **https://github.com/nomiclabs/hardhat/issues/1117#issuecomment-779213915** here, finally got past the few-hour debugging headache with this!-)

**Comment by vedant-11** [comment]
for me npx hardhat clean worked

**Comment by jikdo** [comment]
`npx hardhat clean` works  , wish i knew how this error is thrown though

**Comment by fvictorio** [comment]
Copying here what @morgansliman found out in #2507:

> The core problem being fixed is that when we change the contract name (or presumably make any change to the contract), the verification pipeline recompiles only that contract and not `console.sol`.
This results in the existing artifact debug file for `console.sol` pointing to the old hash of the `contracts/Greeter.sol:Greeter` build-info and not the newly recompiled `contracts/Greeter.sol:Foo` hash and causing the error from the original issue.

**Comment by asardon** [comment]
run into the same issue, thank you @SirLancelot-OG!

**Comment by SomeJoker** [comment]
> same

Nice. fix my problem

**Comment by jacob0123456** [comment]
I used npx hardhat clean, but I hadn't solved the problem.
Please help me.

**Comment by alvinChristianto** [comment]
Thankyou that save me !

**Comment by fvictorio** [comment]
We finally fixed this in [Hardhat v2.17.0](https://github.com/NomicFoundation/hardhat/releases/tag/hardhat%402.17.0). The `npx hardhat clean` workaround should no longer be necessary.

If someone is using v2.17.0 or later and runs into this problem, please let us know!
